### ISSUE TITLE
Improve stacktraces for required modules

### ISSUE THREAD

**Issue by selaux** [cgjs/cgjs#6]
Hi,

I was trying out cgjs with my [gtk react renderer](https://github.com/selaux/react-gtk/tree/cgjs). I worked almost out of the box. One of the things that bother me though, is that stacktraces all point to the `require.js` file like this:

```
MyApp@/react-gtk/node_modules/cgjs/cgjs/require.js line 56 > Function:17:15
constructClassInstance@/react-gtk/node_modules/cgjs/cgjs/require.js line 56 > Function:2141:20
updateClassComponent@/react-gtk/node_modules/cgjs/cgjs/require.js line 56 > Function:3625:9
beginWork@/react-gtk/node_modules/cgjs/cgjs/require.js line 56 > Function:4011:16
performUnitOfWork@/react-gtk/node_modules/cgjs/cgjs/require.js line 56 > Function:6010:16
workLoop@/react-gtk/node_modules/cgjs/cgjs/require.js line 56 > Function:6074:26
anonymous/module.exports/invokeGuardedCallback$1@/react-gtk/node_modules/cgjs/cgjs/require.js line 56 > Function:1562:5
anonymous/module.exports/ReactErrorUtils.invokeGuardedCallback@/react-gtk/node_modules/cgjs/cgjs/require.js line 56 > Function:1509:5
renderRoot@/react-gtk/node_modules/cgjs/cgjs/require.js line 56 > Function:6152:7
performWorkOnRoot@/react-gtk/node_modules/cgjs/cgjs/require.js line 56 > Function:6800:24
performWork@/react-gtk/node_modules/cgjs/cgjs/require.js line 56 > Function:6753:7
requestWork@/react-gtk/node_modules/cgjs/cgjs/require.js line 56 > Function:6664:7
scheduleWorkImpl@/react-gtk/node_modules/cgjs/cgjs/require.js line 56 > Function:6518:11
scheduleWork@/react-gtk/node_modules/cgjs/cgjs/require.js line 56 > Function:6475:12
scheduleTopLevelUpdate@/react-gtk/node_modules/cgjs/cgjs/require.js line 56 > Function:6979:5
anonymous/module.exports/ReactFiberReconciler/<.updateContainer@/react-gtk/node_modules/cgjs/cgjs/require.js line 56 > Function:7017:7
render@/react-gtk/node_modules/cgjs/cgjs/require.js line 56 > Function:16:13
anonymous/<@/react-gtk/node_modules/cgjs/cgjs/require.js line 56 > Function:41:5
anonymous@/react-gtk/node_modules/cgjs/cgjs/require.js line 56 > Function:45:1
require@/react-gtk/node_modules/cgjs/cgjs/require.js:65:15
@/react-gtk/node_modules/cgjs/cgjs/program.js:13:9
@/react-gtk/node_modules/cgjs/cgjs/program.js:6:5
@/react-gtk/node_modules/cgjs/cgjs/program.js:1:1
@/react-gtk/node_modules/.bin/cgjs:67:25
@/react-gtk/node_modules/.bin/cgjs:58:3
@/react-gtk/node_modules/.bin/cgjs:4:2
```

It can get really hard to find the exact function you are looking for this way.

I tried around a little bit by setting the `.name` or `.displayName` of the function that is created from the source of modules in `require.js`, but couldn't achieve anything to change the name in the stacktrace. I think what could work is setting the `.caller` on the function, but that does not work in strict mode. Any ideas?

**Comment by WebReflection** [comment]
to be honest, I think that stack is quite easy to follow

```
MyApp
constructClassInstance
updateClassComponent
beginWork
performUnitOfWork
workLoop
anonymous/module.exports/invokeGuardedCallback$1
anonymous/module.exports/ReactErrorUtils.invokeGuardedCallback
renderRoot
performWorkOnRoot
performWork
requestWork
scheduleWorkImpl
scheduleWork
scheduleTopLevelUpdate
anonymous/module.exports/ReactFiberReconciler/<.updateContainer:7017:7
render
anonymous/<
anonymous
require
```

AFAIK I don't have any control over gjs stack trace so I'm not sure there's anything I can do.

**Comment by selaux** [comment]
Sure I can determine the name of functions from the stack trace, but what if I have 10 files with a `render` function (or a bunch of anonymous functions)? It won't be as easy as in this example.

I know there is limited control over the stack trace, but currently it basically is defined through the implementation of `require.js`. E.g. if `require.js` would somehow use `imports`, the file name would probably be included in the stacktrace. I'm not saying I have a solution though, just that this is something to think about.

**Comment by jaszhix** [comment]
In spidermonkey, you can get the correct path in stack traces by appending `//# sourceURL=${path}` to the JS string.

**Comment by selaux** [comment]
@jaszhix Nice, thx for the hint. Adding this makes the stacktrace render nicely for me. It became:

```
MyApp@/react-gtk/example/index.js:17:15
constructClassInstance@/react-gtk/node_modules/react-reconciler/cjs/react-reconciler.development.js:2141:20
updateClassComponent@/react-gtk/node_modules/react-reconciler/cjs/react-reconciler.development.js:3625:9
beginWork@/react-gtk/node_modules/react-reconciler/cjs/react-reconciler.development.js:4011:16
performUnitOfWork@/react-gtk/node_modules/react-reconciler/cjs/react-reconciler.development.js:6010:16
workLoop@/react-gtk/node_modules/react-reconciler/cjs/react-reconciler.development.js:6074:26
anonymous/module.exports/invokeGuardedCallback$1@/react-gtk/node_modules/react-reconciler/cjs/react-reconciler.development.js:1562:5
anonymous/module.exports/ReactErrorUtils.invokeGuardedCallback@/react-gtk/node_modules/react-reconciler/cjs/react-reconciler.development.js:1509:5
renderRoot@/react-gtk/node_modules/react-reconciler/cjs/react-reconciler.development.js:6152:7
performWorkOnRoot@/react-gtk/node_modules/react-reconciler/cjs/react-reconciler.development.js:6800:24
performWork@/react-gtk/node_modules/react-reconciler/cjs/react-reconciler.development.js:6753:7
requestWork@/react-gtk/node_modules/react-reconciler/cjs/react-reconciler.development.js:6664:7
scheduleWorkImpl@/react-gtk/node_modules/react-reconciler/cjs/react-reconciler.development.js:6518:11
scheduleWork@/react-gtk/node_modules/react-reconciler/cjs/react-reconciler.development.js:6475:12
scheduleTopLevelUpdate@/react-gtk/node_modules/react-reconciler/cjs/react-reconciler.development.js:6979:5
anonymous/module.exports/ReactFiberReconciler/<.updateContainer@/react-gtk/node_modules/react-reconciler/cjs/react-reconciler.development.js:7017:7
render@/react-gtk/src/renderer.js:16:13
anonymous/<@/react-gtk/example/index.js:41:5
anonymous@/react-gtk/example/index.js:45:1
require@/react-gtk/node_modules/cgjs/cgjs/require.js:60:15
@/react-gtk/node_modules/cgjs/cgjs/program.js:12:9
@/react-gtk/node_modules/cgjs/cgjs/program.js:5:5
@/react-gtk/node_modules/cgjs/cgjs/program.js:1:1
@/react-gtk/node_modules/.bin/cgjs:67:25
@/react-gtk/node_modules/.bin/cgjs:58:3
@/react-gtk/node_modules/.bin/cgjs:4:2
```
I will create a PR.

**Comment by WebReflection** [comment]
fixed in 0.1.25. thank you both !!!
